### PR TITLE
Added support for Jacoco XML parsing

### PR DESCRIPTION
**Pull request by leandrogianotti**
### **PR Type**
Enhancement, Bug fix

___

### **Description**
- Added support for parsing JaCoCo XML coverage reports by implementing the `parse_missed_covered_lines_jacoco_xml` method.
- Introduced the `get_file_extension` method to determine the file extension of the coverage report.
- Refactored the `parse_coverage_report_jacoco` method to handle both XML and CSV formats, raising an error for unsupported formats.
- Improved error handling and code readability.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>CoverageProcessor.py</strong><dd><code>Add support for JaCoCo XML parsing and refactor coverage processing</code></dd></summary>
<hr>

cover_agent/CoverageProcessor.py

<li>Added support for parsing JaCoCo XML coverage reports.<br> <li> Introduced a method to determine file extension.<br> <li> Refactored code to handle different file formats (XML, CSV).<br> <li> Improved error handling for unsupported formats.<br>

</details>

  </td>
  <td><a href="https://github.com/Codium-ai/cover-agent/pull/132/files#diff-610c623a5c322b67c9a12fa75021723cc335d4dc3ce4676eb8576a20f3ffd93e">+42/-4</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Review by barnett-yuxiang — COMMENTED** [review]
its ok

**Comment by EmbeddedDevops1** [comment]
Looks like the CI job is failing. Unit tests are not currently passing.

File: cover_agent/CoverageProcessor.py [hunk 1]
```diff
@@ -222,3 +252,11 @@ def extract_package_and_class_java(self):
...
 
         return package_name, class_name
+
+    def get_file_extension(self, filename: str) -> str | None:
+        """Get the file extension from a given filename."""
+        match = re.search(r'\.(\w+)$', filename)
+        if match:
+            return match.group(1)
+        else:
+            return None
```
Comment (EmbeddedDevops1) [inline cover_agent/CoverageProcessor.py:262]: I'm pretty sure you can just use an `os` (or similar) library to extract out a path name. No need for this function.

**Comment by EmbeddedDevops1** [comment]
@leandrogianotti Thanks again for the PR. Are you able to add a unit test that validates the XML piece of the coverage parser?

**Comment by EmbeddedDevops1** [comment]
@leandrogianotti This is great and thanks for adding the unit tests. Did you run this against any of the templated projects (or your own) to validate XML parsing is working as expected?

**Comment by leandrogianotti** [comment]
> @leandrogianotti This is great and thanks for adding the unit tests. Did you run this against any of the templated projects (or your own) to validate XML parsing is working as expected?

I have run it against my own project which by configuration only generates the Jacoco report in xml and it worked fine. Do you need me to run it against the project templates and attach evidence?

**Comment by EmbeddedDevops1** [comment]
That would be ideal just so we have the validation.

**Review by EmbeddedDevops1 — APPROVED** [review]

**Pull request merged**